### PR TITLE
session: boostrap delete isolation engines from global variables

### LINKED ISSUE #16624: tidb-server upgrade from 3.1-beta to 4.0-rc, but there's no any data in tidb-dashboard
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
https://asktug.com/t/topic/33658/23
use tiup upgrade tidb-server from 3.1-beta to 4.0-rc

### 2. What did you expect to see?
All the things are work normally.

### 3. What did you see instead?
There's none data in tidb-dashboard.

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)

tidb 4.0-rc

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve?
issue: close #16624 

Problem Summary: When tidb-server is upgraded from 3.1-beta to a higher version. There is a `tidb_isolation_read_engines` in `mysql.global_variables`. It may lead a wrong behavior such as tidb-dashboard can't read the cluster-info table. So we add a new version for bootstrap to delete it.

### What is changed and how it works?

How it Works: delete the value in `mysql.global_variables` when tidb-server bootstrap.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU

### CODE REVIEW

File: session/bootstrap.go [hunk 1]
```diff
@@ -441,170 +489,17 @@ func upgrade(s Session) {
...
-
-	if ver < version27 {
-		upgradeToVer27(s)
-	}
-
-	if ver < version28 {
-		upgradeToVer28(s)
+	for version, upgradeFunc := range bootstrapVersion {
+		if ver < version {
+			upgradeFunc(s)
```
Comment (zz-jason) [inline session/bootstrap.go:494]: should we update to these versions in ascending order from version 2 to version 43?

Comment (lzmhhh123) [reply session/bootstrap.go]: done

File: session/bootstrap.go [hunk 2]
```diff
@@ -441,170 +490,22 @@ func upgrade(s Session) {
...
-	}
-
-	if ver < version26 {
-		upgradeToVer26(s)
-	}
-
-	if ver < version27 {
-		upgradeToVer27(s)
+	versionsUpgrade := make([]int64, 0, len(bootstrapVersion))
+	for version := range bootstrapVersion {
```
Comment (tiancaiamao) [inline session/bootstrap.go:494]: The range operation does not keep order as the map key!

File: session/bootstrap.go [hunk 3]
```diff
@@ -441,170 +490,22 @@ func upgrade(s Session) {
...
+		if ver < version {
+			versionsUpgrade = append(versionsUpgrade, version)
```
Comment (tiancaiamao) [inline session/bootstrap.go:496]: ver = version?

File: session/bootstrap.go [hunk 4]
```diff
@@ -377,6 +378,54 @@ const (
 	version41 = 41
 	// version42 add storeType and reason column in expr_pushdown_blacklist
 	version42 = 42
+	// version43 delete tidb_isolation_read_engines from mysql.global_variables to avoid unexpected behavior after upgrade.
+	version43 = 43
+)
+
+var (
+	bootstrapVersion = map[int64]func(Session){
```
Comment (tiancaiamao) [inline session/bootstrap.go:386]: Why not use a slice?
`[]func(Session)` or `[]struct{fn Session, ver int}`

**Pull request merged**